### PR TITLE
Fixed #27430 -- Added -b/--buffer option to DiscoverRunner.

### PR DESCRIPTION
**Pull request by bmispelon**
I couldn't figure out how to make it work with parallel test execution

ticket-27430

### CODE REVIEW

File: docs/releases/3.1.txt [hunk 1]
```diff
@@ -176,7 +176,8 @@ Logging
 Management Commands
 ~~~~~~~~~~~~~~~~~~~
 
-* ...
+* Django test runner now supports a ``--buffer``/``-b`` option to discard
+  output on passing tests.
```
Comment (felixxm) [inline docs/releases/3.1.txt]: Please move it under the `Tests` section.

File: docs/ref/django-admin.txt [hunk 2]
```diff
@@ -1468,6 +1468,13 @@ Runs test methods and classes matching test name patterns, in the same way as
 Spawns a ``pdb`` debugger at each test error or failure. If you have it
 installed, ``ipdb`` is used instead.
 
+.. django-admin-option:: --buffer, -b
+
+.. versionadded:: 3.1
+
+Discard output (stdout and stderr) for passing tests, in the same way as
```
Comment (felixxm) [inline docs/ref/django-admin.txt]: `Discard` -> `Discards`

Comment (felixxm) [reply docs/ref/django-admin.txt]: I think we can add backticks to ``` ``stdout`` ``` and ``` ``stderr`` ```.

File: django/test/runner.py [hunk 3]
```diff
@@ -451,6 +451,9 @@ def __init__(self, pattern=None, top_level=None, verbosity=1,
         self.pdb = pdb
         if self.pdb and self.parallel > 1:
             raise ValueError('You cannot use --pdb with parallel tests; pass --parallel=1 to use it.')
+        self.buffer = buffer
+        if self.buffer and self.parallel > 1:
+            raise ValueError('You cannot use -b/--buffer with parallel tests; pass --parallel=1 to use it.')
```
Comment (felixxm) [inline django/test/runner.py]: Please wrap at 79 chars:
```python
raise ValueError(
    'You cannot use -b/--buffer with parallel tests; pass '
    '--parallel=1 to use it.'
)
```

Comment (felixxm) [reply django/test/runner.py]: We should add test for raising an error, e.g. #12194.

**Comment by bmispelon** [comment]
Thanks for the review!

I've made the changes you requested (in separate commits, I'm not if you prefer that or a whole squahsed commit).
I've also rebased on the latest master to resolve the conflicts with the test you added for `--pdb + --parallel`.

**Comment by felixxm** [comment]
@bmispelon Thanks :+1: I pushed minor edits and simplified tests.

**Comment by bmispelon** [comment]
Thanks! :sparkles:

**Pull request merged**